src/runtime/volatile/register_nxpmk66f18.go

```diff
@@ -0,0 +1,79 @@
+// +build nxp,mk66f18
+
+package volatile
+
+import "unsafe"
+
+const registerBase = 0x40000000
+const registerEnd = 0x40100000
+const bitbandBase = 0x42000000
+const ptrBytes = unsafe.Sizeof(uintptr(0))
+
+//go:inline
+func bitbandAddress(reg uintptr, bit uint8) uintptr {
+	if uintptr(bit) > ptrBytes*8 {
+		panic("invalid bit position")
+	}
+	if reg < registerBase || reg >= registerEnd {
+		panic("register is out of range")
+	}
+	return (reg-registerBase)*ptrBytes*8 + uintptr(bit)*ptrBytes + bitbandBase
```

Comment:
> Because this feature is specific to the Cortex-M series microcontrollers

Do you have suggestions on making this file usable for other Cortex-M's and not just the K66? I'm guessing they all do essentially the same math, but `registerBase`, `registerEnd`, and `bitbandBase` seem like they could be processor-specific.

I could change the build tags to `cortexm,bitband` so that targets could opt in, and then require them to define those constants. Except they probably should be defined in `runtime`, which means they would need to be exported.